helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -243,6 +243,18 @@ public Response updateCluster(@PathParam("clusterId") String clusterId,
         }
         break;
 
+      case deactivate:
+        if (superCluster == null) {
+          return badRequest("Super Cluster name is missing!");
+        }
+        try {
+          clusterSetup.activateCluster(clusterId, superCluster, false);
+        } catch (Exception ex) {
+          LOG.error("Failed to deactivate cluster {} from super cluster {}.", clusterId, superCluster);
+          return serverError(ex);
```

Comment:
Let's not use server error. It means Helix REST service has issue. But this is actually there is no super cluster. It should be invalid request or something.

Reply:
There could be server issues that caused the Exception other than illegal argument. To my understanding, a better fix is we should throw and catch different types of exception, and handle respectively on REST layer. 

Note that this is the same pattern as in `activate` API. I think the fix above is a little out of scope. We can create a separate issue to track this and cleanup later. Let me know how you think. Thanks.